# [\#8231 PR](https://github.com/ethereum/solidity/pull/8231) `merged`:  Fix CompilerStack::loadMissingSources()

#### <img src="https://avatars.githubusercontent.com/u/5008794?u=aa5f725afdad81154a79cd5ab6be9340b08da4a9&v=4" width="50">[aarlt](https://github.com/aarlt) opened issue at [2020-02-04 00:16](https://github.com/ethereum/solidity/pull/8231):

An exception was not caught within `CompilerStack::loadMissingSources()`, if more than 256 errors where detected. 

The main problem related to #8102 was that the used standard-json tried to import a lot of additional `*.sol` files that where not present. These errors exceeded the 256 error count limit and finally ended up with an `FatalError` exception. I added an additional test `test/libsolidity/syntaxTests/more_than_256_importerrors.sol` to cover those kind of errors. 

#### <img src="https://avatars.githubusercontent.com/u/9073706?v=4" width="50">[chriseth](https://github.com/chriseth) commented at [2020-02-04 15:40](https://github.com/ethereum/solidity/pull/8231#issuecomment-581969406):

Warnings were treated differently than errors because we continue after 256 warnings. For errors we really abort. The difference is that compilation can still work with just warnings, but it failed already with at least one error. Please try to see if the fatal error exception is just not caught properly.

#### <img src="https://avatars.githubusercontent.com/u/5008794?u=aa5f725afdad81154a79cd5ab6be9340b08da4a9&v=4" width="50">[aarlt](https://github.com/aarlt) commented at [2020-02-04 15:49](https://github.com/ethereum/solidity/pull/8231#issuecomment-581974086):

Ok I will look into how we can do the abort.

#### <img src="https://avatars.githubusercontent.com/u/9073706?v=4" width="50">[chriseth](https://github.com/chriseth) commented at [2020-02-06 14:19](https://github.com/ethereum/solidity/pull/8231#issuecomment-582926123):

@aarlt what is the status here?

#### <img src="https://avatars.githubusercontent.com/u/5008794?u=aa5f725afdad81154a79cd5ab6be9340b08da4a9&v=4" width="50">[aarlt](https://github.com/aarlt) commented at [2020-02-06 15:29](https://github.com/ethereum/solidity/pull/8231#issuecomment-582959137):

@chriseth I think its ready.

#### <img src="https://avatars.githubusercontent.com/u/5008794?u=aa5f725afdad81154a79cd5ab6be9340b08da4a9&v=4" width="50">[aarlt](https://github.com/aarlt) commented at [2020-02-06 16:27](https://github.com/ethereum/solidity/pull/8231#issuecomment-582987739):

@chriseth hmm.. looked at the code again.. I think its better to move that catch to the parse function

#### <img src="https://avatars.githubusercontent.com/u/9073706?v=4" width="50">[chriseth](https://github.com/chriseth) commented at [2020-02-12 15:57](https://github.com/ethereum/solidity/pull/8231#issuecomment-585274822):

Please squash.


-------------------------------------------------------------------------------



[Export of Github issue for [ethereum/solidity](https://github.com/ethereum/solidity). Generated on 2024.12.15 at 06:45:24.]
